composables/use-subscription.ts

```diff
@@ -0,0 +1,90 @@
+import { PaywallModal } from '#components'
+
+export function useSubscription() {
+  const { t: $t } = useI18n()
+  const { user } = useUserSession()
+  const selectedPlan = ref('yearly')
+  const { loggedIn: hasLoggedIn } = useUserSession()
+  const accountStore = useAccountStore()
+  const isProcessingSubscription = ref(false)
+  const { handleError } = useErrorHandler()
+  const localeRoute = useLocaleRoute()
+  const toast = useToast()
+
+  const isLikerPlus = computed(() => {
+    if (!hasLoggedIn.value) return false
+    return user.value?.isLikerPlus
+  })
+
+  const overlay = useOverlay()
+  const paywallModal = overlay.create(PaywallModal, {
+    props: {
+      'modelValue': toRef(selectedPlan, 'value'),
+      'onUpdate:modelValue': (val: 'monthly' | 'yearly') => {
+        selectedPlan.value = val
+      },
+      'onSubscribe': startSubscription,
+      isProcessingSubscription,
+    },
+  })
+
+  async function checkLikerPlusStatus() {
+    try {
+      if (isLikerPlus.value) {
+        await navigateTo(localeRoute({ name: 'account' }))
+        return true
+      }
+    }
+    catch (error) {
+      await handleError(error)
+    }
+    return false
+  }
```

Comment:
on second thought, after taking out the login, we can further simplify to this, also rename the function for better code understanding
```suggestion
  async function redirectIfSubscribed() {
    if (isLikerPlus.value) {
      await navigateTo(localeRoute({ name: 'account' }))
      return true
    }
    return false
  }
```

Reply:
[♻️ Simplify isSubscribed check](https://github.com/likecoin/3ook-com/pull/57/commits/5cfbd311c8a32e839bc506f7f8d06dc932628e24)